File: dcp-solutions/2024-10-30/solution.go
```markdown
package main

import (
    "fmt"
)

func shuffleDeck(deck []int) {
    n := len(deck)
    for i := n - 1; i > 0; i-- {
        j := randInt(i, n-1)
        deck[i], deck[j] = deck[j], deck[i]
    }
}

func randInt(a int, b int) int {
    return a + randIntMod(b-a+1, 999_999_999)
}
func randIntMod(m int, offset int) int {
    return (randInt(0, m-1) + offset) % m
}

func main() {
    var deck = []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12}
    shuffleDeck(deck)
    for i := range deck {
        fmt.Printf("%d ", deck[i])
    }
}
```

File: dcp-solutions/2024-10-30/solution.py
```python
def shuffleDeck(deck, k):
    def rand_index():
        nonlocal seed
        seed = (seed * 3 + 1) % k
        return deck.index(deck[seed])

    deck_copy = deck[:]
    random.seed(seed)
    for i in range(len(deck) - 1, 0, -1):
        j = rand_index()
        deck[i], deck[j] = deck[j], deck[i]
```